docs/design/jet/025-jet-job-partition-pruning.md

```diff
@@ -123,10 +123,71 @@ try to choose only the required members.
 
 ### Scan partition pruning
 
-Partition pruning for IMap scan is a pretty simple optimization: we will extract the partition key 
-condition during SQL opt phase, pass it to a specialized processor supplier which will spawn 
+Key principle of scan partition pruning is that specialized processor meta supplier which will spawn scan processors 
+should be able to self-sufficiently calculate exact set of partitions to scan. To achieve it, we designed a convenient 
+API on a SQL side to be able to pass a minimum required information to PMS constructor.
+
+#### SQL side design details (should be moved to SQL design doc)
+Our goal is to have a **precise** partition set to scan for all prunable `FullScan`-s in resulting plan. 
+To make it possible, and also isolate the implementation for each specific connector, we would like to move 
+the computational process to `SqlConnector`. We extended `fullScanReader` method in `SqlConnector` interface 
+to accept extracted all partition pruning candidates as a parameter and calculate it in connector-specific way:
+```
+    @Override
+    @Nonnull
+    public Vertex fullScanReader(
+            @Nonnull DagBuildContext context,
+            @Nullable HazelcastRexNode predicate,
+            @Nonnull List<HazelcastRexNode> projection,
+            @Nullable List<Map<String, Expression<?>>> partitionPruningCandidates, // <-- new parameter
+            @Nullable FunctionEx<ExpressionEvalContext, EventTimePolicy<JetSqlRow>> eventTimePolicyProvider)
+```
+
+The new parameter `partitionPruningCandidates` is a list of maps, where for each column present in the filter 
+the column name maps to the extracted comparison expression. The connector-specific implementation should
+filter predicates for columns where partitioning strategy is applicable, and check if partitioning strategy 
+is applicable generally, and in case of success transform the input to the **list of inner list of expressions**. 
+Each inner list of expressions contains comparison expressions, and may be **single-element**, if the partitioning 
+strategy key is **simple**, and **multi-element**, if the partitioning strategy key is **composite**.  
+If we have more than one prunable filter predicate, outer list will be multi-element. 
+
+With such design there is a case, when same IMap is used for multiple scans. Self-join and self-union queries are pretty
+uncommon cases, so we can sacrifice precision for simplicity in that case. 
+
+**Currently, it is implemented only for IMap connector, where all expressions are supported.**
+
+For better imagination we prepared an example below.
+
+#### Successful case example (should be moved to SQL design doc)
+Let's assume we have an IMap `map` with composite key {comp1, comp2, comp3} and applied partitioning strategy for
```

Comment:
this example assumes AttributePartitionStrategy, currently there is no other strategy that would fit it, but let's be explicit